ci/publish_github_release.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# NOTE: please do not set trace mode (-x)
+# as it will leak credentials in the logs.
+set -e
+
+if [[ ! -f "${ENVOY_SRCDIR}/build_release_stripped/envoy" ]]; then
+    echo "could not locate envoy binary at path: ${ENVOY_SRCDIR}/build_release_stripped/envoy"
+    exit 1
+fi
+
+if [[ -z "${GITHUB_TOKEN:-}" ]]; then
+    echo "environment variable GITHUB_TOKEN unset; cannot continue with publishing."
+    
+    # TODO(taion809): discuss whether or not failing to publish to github warrents failing the build itself
+    exit 0
```

Comment:
Yeah, please make these all `exit 1`, I think if we don't have the expected envs, then something has gone badly wrong and we should be notified early.